### ISSUE TITLE
Review of February 2020 development

### ISSUE THREAD

**Issue by arni-magnusson** [r-gregmisc/gplots#1]
Hi Greg,

Below is a summary of my [39 commits](https://github.com/r-gregmisc/gplots/commits/master) in February, for your review. Some are clear improvements (tag CRAN releases, fix typos, add bubbleplot) while others were perhaps not quite as necessary, but still improvements I hope (comb through Rd files to standardize format, etc.).

- [x] 8e3c49f CRAN version 3.0.1.1.
- [x] 6293b7c CRAN version 3.0.1.2.
- [x] 14adbe7 Merge CRAN and warnes/r-gregmisc/gplots
- [x] cb708e2, 6fc42c0, e1459d4 Link to online Balloon Plot paper, rather than including PDF inside inst folder.
- [x] 8ffdc83 Consistent indentation and format of balloonplot.R code.
- [x] 3f7df9d, e1459d4 List of functions is now in ?gplots, same as help("gplots-package"), as well as on the front page [readme](https://github.com/r-gregmisc/gplots). Once on CRAN, we can link from the readme to the online help page, so we maintain list in one place.
- [x] 60d0201 Alternative format for NEWS file, so news(package="gplots") now displays as HTML in a browser.
- [x] 882d317, 1374128, 32b6749, 73b9a5b, 29e1bcf, d3401ee, 1aa09ff, 20ca6fd, 3d9b121, 1d4a3dd, 885f083 Many small improvements in Rd files, consistent style similar to core R.
- [x] 3612a18 Consistent indentation and format of heatmap.2.R code.
- [x] e8b85d9 Add bubbleplot.
- [x] 53b972c Remove header comments.
- [x] ed28596 Remove gdata dependency. gplots now depends on gtools, caTools, and bitops.

In general, I resisted the temptation to edit *.R files, except whitespace and comments in balloonplot.R and heatmap.2.R. My main effort was to make small improvements to the help pages, both from the user and maintainer perspective. I don't feel strongly about any of these commits, so feel free to make any additional commits that revert some of those changes.

As far as I'm concerned, the package is ready for a 3.1.0 CRAN release, to remove the orphaned status. My future edits to gplots will be much smaller and easier to review. I hope this 'spring cleaning' is well received :)

**Comment by warnes** [comment]
FWIW, I would be happy to see gplots converted to use ROxygen instead of .Rd files.  Perhaps using https://yihui.org/rd2roxygen...

**Comment by arni-magnusson** [comment]
Thanks for the thorough ongoing review, Greg. Lots of good points that I agree with.

I'll do the edits on bubbleplot and you can edit everything else directly. No need for me to review your review :)

**Comment by warnes** [comment]
I've finished my review.

Thank you for all of the hard work!!! 

Would you mind doing the edits?  I've used up (more than) the time alotted...

**Comment by arni-magnusson** [comment]
No problem, will do!

**Comment by arni-magnusson** [comment]
List of remaining discussion points:
- [ ] ["aut" vs. "ctb"](https://github.com/r-gregmisc/gplots/commit/60d0201d72c241c60880dae41ec8de6c71696a71#r37481342)

**Comment by arni-magnusson** [comment]
Hi Greg, I've processed your review. Thanks for the dedication to go through all of the February commits!

You found several changes of mine that were not that great, but instead of reverting them you came up with even better solutions. What remains is a much improved package: a new function, fewer package dependencies, a package help page, and tidier help pages with a consistent (base R) style.

If you `git checkout master`, `git merge arni` and `git push`, I will probably just delete the arni branch. I can create a new feature branch in the future, if I want to make a specific suggestion.

I think we can also `git checkout develop`, `git merge master` and `git push` to fast-forward develop to catch up with master. Then all branches would be even, ready for Tal's additions. Should we make the master branch the default branch?

P.S. The https://github.com/r-gregmisc/gplots/commit/60d0201d72c241c60880dae41ec8de6c71696a71#r37492486 question is still pending.

**Comment by warnes** [comment]
Hi Arni, 

The normal way to handle changes is to make them on a branch, then create a pull request (PR), which provides a clean mechanism for reviewing and commenting on the changes.  See the official docs [here](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-requests).

When the PR is merged, it includes an option to delete the old branch.

Can you [create a PR](https://github.com/r-gregmisc/gplots/compare/develop...arni) for the arni branch?  I'll then do a final review and approve, after which any of use can do the merge. 

I really appreciate all of your hard work!

-G

**Comment by arni-magnusson** [comment]
I agree, pull requests are the way to go. I have now created pull request #5 and issue #6. We can either close issue #1 now, or when these have been resolved:
#5 (Pull Request) Revisions from February 2020 development review
#6 (Issue) Update list of authors/contributors
